src/web/components/ApiKeyManagement/keyEditDialog.spec.tsx

```diff
@@ -0,0 +1,267 @@
+/* eslint-disable camelcase */
+import { composeStories } from '@storybook/testing-react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+
+import KeyEditDialog, {  } from './KeyEditDialog';
+import * as stories from './KeyEditDialog.stories';
+
+const { KeyWithRolesParticipantIsntAllowed, MultipleRoles } = composeStories(stories);
+
+describe('Key edit dialog', () => {
+  it('has the correct title', () => {
+    render(<MultipleRoles />);
+
+    const openButton = screen.getByText('Open');
+    fireEvent.click(openButton);
+
+    expect(screen.getByText('Edit ApiKey')).toBeInTheDocument();
+  });
+
+  it('should prefill name input', () => {
+    render(<MultipleRoles />);
+
+    const openButton = screen.getByText('Open');
+    fireEvent.click(openButton);
+
+    expect(screen.getByRole('textbox', { name: 'newName' })).toHaveValue('ApiKey');
+  });
+
+  it('should select the current rows', () => {
+    render(<MultipleRoles />);
+
+    const openButton = screen.getByText('Open');
+    fireEvent.click(openButton);
+
+    for (const role of ['ID_READER', 'SHARER']) {
+      expect(screen.getByDisplayValue(role)).not.toBeChecked();
+    }
+
+    for (const role of ['MAPPER', 'GENERATOR']) {
+      expect(screen.getByDisplayValue(role)).toBeChecked();
+    }
+  });
+
+  it('should show both the keys and allowed roles', () => {
```

Comment:
"should show both the keys and allowed roles" -> "should show both the key's roles and the participant's allowed API roles"?